x/recovery/keeper/ibc_callbacks_integration_suite_test.go

```diff
@@ -0,0 +1,334 @@
+						// Relay both packets that were sent in the ibc_callback
+						err := s.pathOsmosisEvmos.RelayPacket(CreatePacket("10000", "aevmos", sender, receiver, "transfer", "channel-0", "transfer", "channel-0", 1, timeout))
+						s.Require().NoError(err)
+						err = s.pathOsmosisEvmos.RelayPacket(CreatePacket("10", "transfer/channel-0/uosmo", sender, receiver, "transfer", "channel-0", "transfer", "channel-0", 2, timeout))
+						s.Require().NoError(err)
+
+						claimAfter, _ := s.EvmosChain.App.(*app.Evmos).ClaimsKeeper.GetClaimsRecord(s.EvmosChain.GetContext(), senderAcc)
+						Expect(claim).To(Equal(claimAfter))
+					})
+				})
+			})
+
+			Context("without a sender's claims record", func() {
+				When("recipient has no ibc vouchers that originated from other chains", func() {
+
+					It("should transfer and recover tokens", func() {
+						// aevmos & ibc tokens that originated from the sender's chain
+						s.SendAndReceiveMessage(s.pathOsmosisEvmos, s.IBCOsmosisChain, coinOsmo.Denom, coinOsmo.Amount.Int64(), sender, receiver, 1)
+						timeout = uint64(s.EvmosChain.GetContext().BlockTime().Add(time.Hour * 4).Add(time.Second * -20).UnixNano())
+
+						// Escrow before relaying packets
+						balanceEscrow := s.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(s.EvmosChain.GetContext(), transfertypes.GetEscrowAddress("transfer", "channel-0"), "aevmos")
+						Expect(balanceEscrow).To(Equal(coinEvmos))
+						ibcOsmo := s.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(s.EvmosChain.GetContext(), receiverAcc, uosmoIbcdenom)
+						Expect(ibcOsmo.IsZero()).To(BeTrue())
+
+						// Relay both packets that were sent in the ibc_callback
+						err := s.pathOsmosisEvmos.RelayPacket(CreatePacket("10000", "aevmos", sender, receiver, "transfer", "channel-0", "transfer", "channel-0", 1, timeout))
+						s.Require().NoError(err)
+						err = s.pathOsmosisEvmos.RelayPacket(CreatePacket("10", "transfer/channel-0/uosmo", sender, receiver, "transfer", "channel-0", "transfer", "channel-0", 2, timeout))
+						s.Require().NoError(err)
+
+						// Check that the aevmos were recovered
+						nativeEvmos := s.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(s.EvmosChain.GetContext(), senderAcc, "aevmos")
+						Expect(nativeEvmos.IsZero()).To(BeTrue())
+						ibcEvmos := s.IBCOsmosisChain.GetSimApp().BankKeeper.GetBalance(s.IBCOsmosisChain.GetContext(), receiverAcc, aevmosIbcdenom)
+						Expect(ibcEvmos).To(Equal(sdk.NewCoin(aevmosIbcdenom, coinEvmos.Amount)))
+
+						// Check that the uosmo were recovered
+						ibcOsmo = s.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(s.EvmosChain.GetContext(), receiverAcc, uosmoIbcdenom)
+						Expect(ibcOsmo.IsZero()).To(BeTrue())
+						nativeOsmo := s.IBCOsmosisChain.GetSimApp().BankKeeper.GetBalance(s.IBCOsmosisChain.GetContext(), receiverAcc, "uosmo")
+						Expect(nativeOsmo).To(Equal(coinOsmo))
+					})
+				})
+
+				// Do not recover uatom that was from Atom through IBC transfer from Osmosis
```

Comment:
// Do not recover uatom sent from Cosmos when performing recovery through IBC transfer from Osmosis